### PR TITLE
doc: suggestions for design page

### PR DESCRIPTION
**Pull request by danbev**
While reading the design page I came across a few sentences that could
perhaps be improved, and this commit contains suggestions for such
improvements.

### CODE REVIEW

File: docs/src/design.rst [hunk 1]
```diff
@@ -77,19 +78,19 @@ stages of a loop iteration:
 #. Prepare handle callbacks are called. Prepare handles get their callbacks called right before
    the loop will block for I/O.
 
-#. Poll timeout is calculated. Before blocking for I/O the loop calculates for how long it should
+#. Poll timeout is calculated. Before blocking for I/O the loop calculates the duration it should
```
Comment (santigimeno) [inline docs/src/design.rst]: I'm not sure this change improves the current text

File: docs/src/design.rst [hunk 2]
```diff
@@ -77,19 +78,19 @@ stages of a loop iteration:
...
    block. These are the rules when calculating the timeout:
 
         * If the loop was run with the ``UV_RUN_NOWAIT`` flag, the timeout is 0.
         * If the loop is going to be stopped (:c:func:`uv_stop` was called), the timeout is 0.
         * If there are no active handles or requests, the timeout is 0.
         * If there are any idle handles active, the timeout is 0.
         * If there are any handles pending to be closed, the timeout is 0.
-        * If none of the above cases was matched, the timeout of the closest timer is taken, or
+        * If none of the above cases matched, the timeout of the closest timer is taken, or
```
Comment (santigimeno) [inline docs/src/design.rst]: maybe use present tense as in the rest of rules: `matches` ?

File: docs/src/design.rst [hunk 3]
```diff
@@ -25,9 +25,10 @@ Handles and requests
 libuv provides users with 2 abstractions to work with, in combination with the event loop:
 handles and requests.
 
-Handles represent long-lived objects capable of performing certain operations while active. Some
-examples: a prepare handle gets its callback called once every loop iteration when active, and
-a TCP server handle get its connection callback called every time there is a new connection.
+Handles represent long-lived objects capable of performing certain operations while active. For
+example, a prepare handle gets its callback called once every loop iteration when active. Another
+example, is a TCP server handle that gets its connection callback called every time there is a
```
Comment (santigimeno) [inline docs/src/design.rst]: I'm not sure about this change. Maybe:

`Some examples: a prepare handle gets its callback called once every loop iteration when active; a TCP server handle that gets its connection callback called every time there is a new connection`

Or even:

```
Some examples: 

- A prepare handle gets its callback called once every loop iteration when active.
- A TCP server handle that gets its connection callback called every time there is a new connection.
```

Comment (danbev) [inline docs/src/design.rst]: I went for the your second suggestion. Let me know what you think.

File: docs/src/design.rst (hunk 1 above)
Comment (danbev) [inline docs/src/design.rst]: I've changed this back to the original now.

File: docs/src/design.rst (hunk 2 above)
Comment (danbev) [inline docs/src/design.rst]: Good point, changed this now. Thanks for the review!

File: docs/src/design.rst [hunk 4]
```diff
@@ -25,9 +25,10 @@ Handles and requests
 libuv provides users with 2 abstractions to work with, in combination with the event loop:
 handles and requests.
 
-Handles represent long-lived objects capable of performing certain operations while active. Some
-examples: a prepare handle gets its callback called once every loop iteration when active, and
-a TCP server handle get its connection callback called every time there is a new connection.
+Handles represent long-lived objects capable of performing certain operations while active. Some examples:
+
+- a prepare handle gets its callback called once every loop iteration when active
+- a TCP server handle that gets its connection callback called every time there is a new connection
```
Comment (santigimeno) [inline docs/src/design.rst]: I would start every sentence with upper case and end with a period.

Comment (danbev) [inline docs/src/design.rst]: Fixed that now, thanks

**Comment by santigimeno** [comment]
Landed in https://github.com/libuv/libuv/commit/12f18159cf2f40fe12c62eb7a0ae22b6dd5548d5. Thanks!

**Pull request closed without merging**